concordium-consensus/src/Concordium/GlobalState/Persistent/Account/StructureV1.hs

```diff
@@ -0,0 +1,1393 @@
+            PersistentAccountStakeEnduringBaker
+                { paseBakerRestakeEarnings = restake,
+                  paseBakerInfo = binfoRef,
+                  paseBakerPendingChange = NoChange
+                }
+    newEnduring <- refMake =<< rehashAccountEnduringData ed{paedStake = baker}
+    return $!
+        acc
+            { accountStakedAmount = stake,
+              accountEnduringData = newEnduring
+            }
+
+-- |Add a delegator to an account.
+-- This will replace any existing staking information on the account.
+addDelegator ::
+    (MonadBlobStore m) =>
+    AccountDelegation 'AccountV2 ->
+    PersistentAccount 'AccountV2 ->
+    m (PersistentAccount 'AccountV2)
+addDelegator AccountDelegationV1{..} acc = do
+    let ed = enduringData acc
+    let del =
+            PersistentAccountStakeEnduringDelegator
+                { paseDelegatorId = _delegationIdentity,
+                  paseDelegatorRestakeEarnings = _delegationStakeEarnings,
+                  paseDelegatorTarget = _delegationTarget,
+                  paseDelegatorPendingChange = NoChange
+                }
+    newEnduring <- refMake =<< rehashAccountEnduringData ed{paedStake = del}
+    return $!
+        acc
+            { accountStakedAmount = _delegationStakedAmount,
+              accountEnduringData = newEnduring
+            }
+
+-- |Update the pool info on a baker account.
+-- This MUST only be called with an account that is a baker.
+updateBakerPoolInfo ::
+    (MonadBlobStore m) =>
+    BakerPoolInfoUpdate ->
+    PersistentAccount 'AccountV2 ->
+    m (PersistentAccount 'AccountV2)
+updateBakerPoolInfo upd = updateEnduringData $ \ed -> case paedStake ed of
+    baker@PersistentAccountStakeEnduringBaker{} -> do
+        oldInfo <- refLoad (paseBakerInfo baker)
+        let newInfo = oldInfo & bieBakerPoolInfo %~ applyBakerPoolInfoUpdate upd
+        newInfoRef <- refMake $! newInfo
+        return $! ed{paedStake = baker{paseBakerInfo = newInfoRef}}
+    _ -> error "updateBakerPoolInfo invariant violation: account is not a baker"
+
+
+-- |Set the baker keys on a baker account.
+-- This MUST only be called with an account that is a baker.
+setBakerKeys ::
+    (MonadBlobStore m) =>
+    BakerKeyUpdate ->
+    PersistentAccount 'AccountV2 ->
+    m (PersistentAccount 'AccountV2)
+setBakerKeys upd = updateStake $ \case
+    baker@PersistentAccountStakeEnduringBaker{} -> do
+        oldInfo <- refLoad (paseBakerInfo baker)
+        let newInfo =
+                oldInfo
+                    & bieBakerInfo
+                        %~ ( (bakerAggregationVerifyKey .~ bkuAggregationKey upd)
+                                . (bakerSignatureVerifyKey .~ bkuSignKey upd)
+                                . (bakerElectionVerifyKey .~ bkuElectionKey upd)
+                           )
+        newInfoRef <- refMake $! newInfo
+        return $! baker{paseBakerInfo = newInfoRef}
+    _ -> error "setBakerKeys invariant violation: account is not a baker"
```

Comment:
In principle it would be better for maintainability not to have these catch-all cases and instead list all the current cases.

That makes it harder to screw up if we are adding cases.